Sources/Foundation/Decimal.swift

```diff
@@ -1699,68 +1699,49 @@ public func NSDecimalString(_ dcm: UnsafePointer<Decimal>, _ locale: Any?) -> St
     if decimalSeparator == nil {
         decimalSeparator = "."
     }
+    let decimalSeparatorReversed = String(decimalSeparator.reversed())
 
-    let decimalSeparatorUtf8View = decimalSeparator.utf8
-
-    let ZERO: CChar = 0x30 // ASCII '0' == 0x30
-    let MINUS: CChar = 0x2d // ASCII '-' == 0x2d
+    let DIGITS: [String] = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
 
     // The 39 is from the mantissa being an 128bit, so log10(2^128) is approx 38.5 giving 39 digits max for the mantissa.
     // 128 is the exponent being an Int8.
     // 1 is for sign.
-    // The last 1 is for null terminator.
-    let bufferSize = 39 + 128 + 1 + decimalSeparatorUtf8View.count + 1
+    let maxResultCount = 39 + 128 + 1 + decimalSeparatorReversed.count
```

Comment:
This can be improved slightly. Although the digit count (39) isn't easily calculable
 the `+1` for the minus sign is known if its needed.
 If the exponent is >=0, then the `128` can be replaced directly with exponent since it is just the number of trailing zeros and there is now need for the `decimalSeparatorReversed`